engine/table/src/main/java/io/deephaven/engine/table/impl/util/ShiftCallback.java

```diff
@@ -0,0 +1,8 @@
+//
+// Copyright (c) 2016-2024 Deephaven Data Labs and Patent Pending
+//
+package io.deephaven.engine.table.impl.util;
+
+public interface ShiftCallback {
```

Comment:
I think `ShiftCallback` is unused and should be removed.
We still have `RowSetShiftData.Callback`, which is somewhat used, and should maybe be implemented in place of `ShiftCallback`.
I think whichever survives need a bit of JavaDoc.

Reply:
Still used it appears:
Implemented in the following places:

- ImmutableConstantFloatSource (io.deephaven.engine.table.impl.sources.immutable)
- ArrayBackedColumnSource (io.deephaven.engine.table.impl.sources)
- NullValueColumnSource (io.deephaven.engine.table.impl.sources)
- SingleValueColumnSource (io.deephaven.engine.table.impl.sources)
- ImmutableConstantByteSource (io.deephaven.engine.table.impl.sources.immutable)
- ImmutableConstantShortSource (io.deephaven.engine.table.impl.sources.immutable)
- ImmutableConstantCharSource (io.deephaven.engine.table.impl.sources.immutable)
- ImmutableConstantDoubleSource (io.deephaven.engine.table.impl.sources.immutable)
- ImmutableConstantIntSource (io.deephaven.engine.table.impl.sources.immutable)
- ImmutableConstantLongSource (io.deephaven.engine.table.impl.sources.immutable)
- NanosBasedTimeArraySource (io.deephaven.engine.table.impl.sources)
- ImmutableConstantObjectSource (io.deephaven.engine.table.impl.sources.immutable)
- ImmutableConstantNanosBasedTimeSource (io.deephaven.engine.table.impl.sources.immutable)


Only called from one place:
NanoBasedTimeArraySource.java
```
    @Override
    public void shift(long start, long end, long offset) {
        nanoSource.shift(start, end, offset);
    }
```